### ISSUE TITLE
Unable to use SSL with MySQL back-end

### ISSUE THREAD

**Issue by burnison** [openark/orchestrator#847]
Setting `MySQLOrchestratorUseMutualTLS=false` currently prevents _any_ SSL connection to the MySQL backend server, as it is used to [guard the call](https://github.com/github/orchestrator/blob/master/go/db/db.go#L117) to `SetupMySQLOrchetsratorTLS`. Further, `SetupMySQLOrchestratorTLS` does not, itself, [guard](https://github.com/github/orchestrator/blob/master/go/db/tls.go#L133) on this same flag and always attempts to add the client key pair.

As the documentation suggests this flag may be used to enable encryption while not requiring client authentication, this is unexpected behaviour. We have a working fix for this problem by additionally evaluating whether or not `MySQLOrchestratorSSLCAFile` is blank, but I'd like to make sure this is not an intentional decision before opening a pull request.

**Comment by shlomi-noach** [comment]
I am not aware this was an intentional decision. All TLS code was added by way of contributions, so I cannot speak on behalf of the contributors.
I'm not sure I understand the nature of the problem: 

> currently prevents any SSL connection to the MySQL backend server

Did you mean any `orchestrator` SSL connection to the MySQL backend server, or that this somehow affects other clients? (which would be super unexpected)

I'm happy to receive a PR.

**Comment by burnison** [comment]
> Did you mean any orchestrator SSL connection to the MySQL backend server, or that this somehow affects other clients? (which would be super unexpected)

This is specifically connections to the MySQL backend server, yes. In particular, SSL _can_ be used _iff_ a client key pair is provided.

**Comment by shlomi-noach** [comment]
> This is specifically connections to the MySQL backend server, yes. In particular, SSL can be used iff a client key pair is provided.

Sorry, I'll ask again since this does not answer my question. Did you mean that because of `orchestrator`'s configuration, _other clients_ which are unrelated to `orchestrator` are unable to connect to the same backend server?

**Comment by burnison** [comment]
No, this only about Orchestrator. _Other_ clients should be out of scope here and can connect, provided correct configuration. When Orchestrator is configured with `MySQLOrchestratorUseMutualTLS=false`, Orchestrator is unable to access its MySQL backend specified in `MySQLOrchestratorHost` using SSL. Sorry for the confusion.
